server/src/query/executeQuery.js

```diff
@@ -26,12 +27,20 @@ async function executeQuery(
     debug('Executing query: ' + query.toString())
     const rawResult = await query
     let result = rawResult
+    let totalCount = result.length
+
+    if (pager) {
+        const countQuery = pager.getTotalCountQuery(query)
```

Comment:
Am I understanding this right in that we are running the same query twice:
1. With a `limit` and `offset`, and whose results are what we return
2. Without `limit` and `offset` but with `count`

Could we instead run it once without `limit` and `offset` and apply those ourselves afterwards?

```js
const applyOffset = (result, offset) => {
  return result.slice(offset)
}

const applyLimit = (result, limit) => {
  return result.slice(0, limit)
}

const result = await query
return {
  count: result.length,
  result: applyOffsetAndLimit(result, { offset, limit })
}
```

Reply:
Yes, because we are doing the offset in the SQL-query. Of course we could do this in JS, and we do have that implemented in the decorated `pagination`-implementation. However that means we need all results in memory. It's a tradeoff, and I think in our case the result-set should stay small enough that it probably doesn't really matter. But I do think the memory usage would become a problem before a second query would.

I can implement a method for this in the Pager-class, and then we can decide between both for different use cases.

Reply:
No need for a separate method I was just wondering what the tradeoffs were :+1: